[inductor] Use decomposition for _to_copy

desertfire opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #90314

Summary: also contains a fix for https://github.com/pytorch/pytorch/issues/89633

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10

desertfire committed:
[inductor] Use decomposition for _to_copy

[ghstack-poisoned]

ngimel commented:
Can you also sneak a fix for #89633 here, when dtype/device/memory format are not specified just call clone.

desertfire committed:
Update on "[inductor] Use decomposition for _to_copy"


Summary: also contains a fix for https://github.com/pytorch/pytorch/issues/89633

[ghstack-poisoned]

desertfire committed:
Update on "[inductor] Use decomposition for _to_copy"


Summary: also contains a fix for https://github.com/pytorch/pytorch/issues/89633

[ghstack-poisoned]

desertfire committed:
Update on "[inductor] Use decomposition for _to_copy"


Summary: also contains a fix for https://github.com/pytorch/pytorch/issues/89633

[ghstack-poisoned]

desertfire committed:
Update on "[inductor] Use decomposition for _to_copy"


Summary: also contains a fix for https://github.com/pytorch/pytorch/issues/89633

[ghstack-poisoned]

desertfire committed:
Update on "[inductor] Use decomposition for _to_copy"


Summary: also contains a fix for https://github.com/pytorch/pytorch/issues/89633

[ghstack-poisoned]

desertfire committed:
Update on "[inductor] Use decomposition for _to_copy"


Summary: also contains a fix for https://github.com/pytorch/pytorch/issues/89633

[ghstack-poisoned]

desertfire committed:
Update on "[inductor] Use decomposition for _to_copy"


Summary: also contains a fix for https://github.com/pytorch/pytorch/issues/89633

cc mlazos soumith voznesenskym yanboliang penguinwu anijain2305 EikanWang jgong5 Guobing-Chen chunyuan-w XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10

[ghstack-poisoned]

desertfire commented:
@pytorchbot merge -f "CI failures are irrelevant"

desertfire commented:
@pytorchbot revert -m "regresses performance on hf_Bert" -c nosignal
